lib/assert.js

```diff
@@ -161,6 +161,14 @@ function isArguments(tag) {
   return tag === '[object Arguments]';
 }
 
+function isMap(object) {
+  return object.constructor === Map;
+}
+
+function isSet(object) {
+  return object.constructor === Set;
+}
```

Comment:
Can you use `isMap` and `isSet` from `process.binding('util')` instead? These checks won’t work for subclasses of Map/Set or Maps/Sets from other VM contexts.